### PR TITLE
Replace mkpasswd with natscli

### LINKED ISSUE #183: Unable to install mkpasswd like in documentation
Hi guys,

I'm trying to install `mkpasswd` from [this documentation](https://docs.nats.io/nats-tools/mkpasswd#installing-mkpasswd).

**Issue No.1** - Wrong symbol in command
Copy button copies content `> go get github.com/nats-io/nats-server/util/mkpasswd` with starting `>`. It makes command invalid and getting error: `-bash: go: Is a directory`

**Issue No.2** - Unable to install mkpasswd

**Steps to reproduce:**
1. Install **go**: `sudo snap install go --classic`
2. Install **mkpasswd**: `go get github.com/nats-io/nats-server/util/mkpasswd`

Actual: getting error
```
~$ go get github.com/nats-io/nats-server/util/mkpasswd
package github.com/nats-io/nats-server/util/mkpasswd: build constraints exclude all Go files in /home/ubuntu/go/src/github.com/nats-io/nats-server/util/mkpasswd
```

Expected: mkpasswd is installed.

OS: 
```Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
```
Go: 
`go version go1.15.6 linux/amd64`

Go env:
```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/ubuntu/.cache/go-build"
GOENV="/home/ubuntu/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/ubuntu/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/ubuntu/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/snap/go/6745"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/snap/go/6745/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build471404296=/tmp/go-build -gno-record-gcc-switches"
```

### PR DESCRIPTION
**Pull request by variadico**
This replaces instances of `mkpasswd` with `natscli`. It also deletes
`mkpasswd.md` and moves some of that content into `natscli.md`.

Resolves #183

### CODE REVIEW

File: nats-tools/natscli.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+For Arch Linux:
+
+```
+> yay natscli
+```
+
+For Docker:
+
+```
+docker pull synadia/nats-server:nightly
```
Comment (ripienaar) [inline nats-tools/natscli.md:27]: It’s in nats-box too. We should probably promote that instead

Comment (variadico) [reply nats-tools/natscli.md]: Cool, I replaced `synadia/nats-server:nightly` with `synadia/nats-box:latest`.

File: nats-tools/natscli.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+For Arch Linux:
+
+```
+> yay natscli
+```
+
+For Docker:
+
+```
+docker pull synadia/nats-box:latest
```
Comment (ColinSullivan1) [inline nats-tools/natscli.md:27]: Might be good to demonstrate how to launch the console.

```suggestion
docker pull synadia/nats-box:latest

docker run -ti synadia/nats-box
```

**Review by ColinSullivan1 — APPROVED** [review]
LGTM, one suggestion.  Thanks!

**Comment by variadico** [comment]
Thanks, Colin! I integrated your change into the PR. 👍🏽

**Pull request merged**